Eric s geo coder

ESmith92 opened the pull request:
Update geocoder to be more generic

-split geocoder class to a geocoder class and url builder class
-split geocoder parser to a parser class and a decoder class
-update spec tests too match new names and smaller classes
-fix white space issues

ESmith92 committed:
google-reverse-geocoder class update

	-add url function
		-move url creation into separate function
		
	-switch curl call over to ken's curl command

ESmith92 committed:
geocoder-results-parser

	-add function definitions
	-improve exeception catching for json string
		-update if to catch any status that is not 'OK'
		-add test for invalid key status
		-add catch for missing 'status' in json string

ESmith92 committed:
fix white space issues

	-fix google-reverse-geocoder
	-geocoder-results-paarser

ESmith92 committed:
fix the test cases for the geocoder classes

	-geocoder-results-paarser
		-remove exception
			-return null on bad json
	
	-google-reverse-geocoder
		-change $this->curl (call to curl class) to $curl_caller

ESmith92 committed:
update geocoder to be more generic

	-geocoder-results-parser
		-rename php file to geocoder-json-decoder
			-remove geocoder-results-parser
		-rename class to JsonDecoder (from GeoParser)
		-move google parser to geocoder-json-parsers
		
	-add geocoder-json-parsers
		-make google default parser
		-move google's parser to geocoder-json-parser
		-add constructor to fill in the decode json object
		-add basic function to select the parser
	
	-add php file geocoder-url-builders
		-make class GeoCoderURLBuilder
			-move Google_URL function to new class
			-add constructor to fill in the geo_coordinates
			-add basic function to select the url builder
			-add variable(s) for url keys when building
			-make google default url builder
			
	-google-reverse-geocoder
		-rename php file to reverse-geocoder
			-remove google-reverse-geocoder
		-rename class to ReverseGeocoder (from GoogleReverseGeocoder)
		-move google_url function to geocoder-url-builders
		-move key variables to geocoder-url-builders
		-update calls to parser, decoder, url builder to the new classes
		-make google the default url builder
		-make google the default json parser

ESmith92 committed:
update geo coder unit test

	-move all into one unit test
	-update class names
		-ReverseGeocoder
		-GeoCoderURLBuilder
		-JsonDecoder
		-GeoJsonParsers

ESmith92 committed:
update geocoder spec test claass names

-GoogleReverseGeoCoderSpec to ReverseGeoCoderSpec 
-GeoParserSpec to GeoJsonParsersSpec

ESmith92 committed:
fix unit test

	-rename class and php file to end with Test
	-fix double name issue in ReverseGeocoder

ESmith92 committed:
Update the geocoder parser spec tests

	-split the spec test to match with the split of the parser and decoder
classes
	
		-GeoJsonParserSpec
			-update class names
			-move decoder tests to JsonDecoderSpec class
			-add default constructor var to tests
			
		-JsonDecoderSpec
			-move decoder tests from GeoJsonParserSpec class
			-update class names

ESmith92 committed:
update reversegeocoderspec test

	-create GeoCoderUrlBuildderSpec class
	-rename ReverseGeocoderSpec to ReverseGecoderClassSpec

ESmith92 committed:
update ReverseGeocoderClassSpec

	-ReverseGeocoderClassSpec
		-remove test for keys (invalid and valid)
		-add tests for results for all combinations of with and without
picking urlbuilder and parser
		-add tests to check default parser and urlbuilder
		
	-ReverseGeocoderClass
		-add functions to get the parser_pick and url_pick varibles for
testing

ESmith92 committed:
update GeoCoderURLBuilderSpec

	-add test for returning a string without selected builder
	-add test for returning a string with google builder selected

ESmith92 committed:
fix errors in spec tests

	-geocoder-url-builders
		-add use to models
		
	-reverse-geocoder-class
		-fix $Url_Pick unfound by adding $this->Url_Pick
		-update url builder class name

ESmith92 committed:
update json parser spec tests

	-add selector tests (return array when using the selector)
	-remove integration test

ESmith92 committed:
update GeoCoderUrlBuilderSpec tests

	-add test to directly test the google url builder
	-fix selector test for google url builder to call the selector function

ESmith92 committed:
fix GeoCoderURLBuilderSpec Tests

	-fix namespace (to spec\SearchApi\Support)
	-fix tests to check for type string

ESmith92 committed:
remove a unneeded comment in geojsonparserspec

ESmith92 committed:
update JsondecoderSpec tests

	-add test to that it returns an array
	-add test to check for exception of json with status unknown

ESmith92 committed:
fix white space issues

ESmith92 committed:
Merge branch 'develop' into EricS-GeoCoder

ESmith92 committed:
remove commented out code

rraub commented on test/spec/SearchApi/Support/GeoCoderURLBuilderSpec.php:
```diff
@@ -0,0 +1,40 @@
+ */
+class GeoCoderURLBuilderSpec extends ObjectBehavior {
+  function it_is_initializable() {
+    $this->beConstructedWith( new Models\GeoCoordinate( 40.714224, -73.961452 ) );
```
you should factor out this test coordinate into a class variable and just reference it by name so we know if it is a valid value or an invalid value that your testing. It mainly helps readability

kenprice committed:
Merge branch 'develop' into EricS-GeoCoder

ESmith92 committed:
move GeoCoordinate variables out of the constructors

	move to new function to improve code readability:
		-move to new function to initialize
		-call function in the constructor
			-just return the coordinates

ESmith92 committed:
Merge branch 'EricS-GeoCoder' of https://github.com/gios-asu/search-api into EricS-GeoCoder

ESmith92 committed:
add constuctor to GeoCoderURLBuilderSpec

	change good_Coord() function to a GeoCoordinate variable and
constructor rather than a function call that returns GeoCoordinate
